app.rb

```diff
@@ -0,0 +1,133 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+ACTIONS = {
+  1 => :list_books,
+  2 => :list_people,
+  3 => :create_person,
+  4 => :create_book,
+  5 => :create_rental,
+  6 => :list_rentals,
+  7 => :break
+}.freeze
+
+class App
+  def print_question
+    puts 'Welcome to School library App!'
+    puts "Please choose an option by entering a number:
+      1- List all books.
+      2- List all people.
+      3- Create a person.
+      4- Create a book.
+      5- Create a rental.
+      6- List all rentals for a given person id.
+      7- Exit."
+  end
+
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  def select_option
+    loop do
+      print_question
+      option = gets.chomp.to_i
+      action = ACTIONS[option]
+
+      if action == :break
+        break
+      elsif action
+        send(action)
+      else
+        puts 'Invalid number, please try again!'
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    num = gets.chomp.to_i
+
+    print 'age:'
+    age = gets.chomp.to_i
+
+    print 'Name:'
+    name = gets.chomp
+
+    case num
+    when 1
+      print 'Has parent permission? [y/n]:'
+      parent_permission = gets.chomp
+      parent_permission = parent_permission == 'y'
+      @person.push(Student.new(age, name, parent_permission: parent_permission))
+    when 2
+      print 'Specialization:'
+      specialization = gets.chomp
+      @person.push(Teacher.new(age, specialization, name))
+    else
+      puts 'Invalid number, please enter number again!'
+    end
+    puts 'Person created successfully 👤'
+  end
+
```

Comment:
when I try to create a person (student or teacher ) I am getting this error
![image](https://github.com/pushkar1995/oop-school-library/assets/87448628/60028cd2-daf7-4010-9e27-c405075c1328)

Kindly fix the arguments error.
Suggestions: 
- you can pass the arguments in the correct order
- the id is not passed nor created in the person